signalfx/notifications.go

```diff
@@ -198,19 +197,36 @@ func validateNotification(val interface{}, key string) (warns []string, errs []e
 			return
 		}
 	case VictorOpsNotificationType:
-		if len(parts) != 3 {
+		if partCount != 3 {
 			errs = append(errs, fmt.Errorf("Invalid VictorOps notification string, please consult the documentation (not enough parts)"))
 			return
 		}
 	case WebhookNotificationType:
-		if len(parts) != 4 {
+		if partCount != 4 {
```

Comment:
should we require 4 parts at all if they provide the credentialId?

like having to do `Webhook,<credentialId>,,` seems silly versus just `Webhook,<credentialId>`.

Though! I understand if we just always want 4 parts and figure things out from there.

Ie, I'm just suggesting moving the `parts[1] != "" ` check "higher", and then check the rest based on that, but I'm ok if that makes things more complicated.

Reply:
one other point of unsolicited feedback :p

this pattern of "4 things in a string but it matters the order" is not very terraform-y; I'd love if some day we had proper structs like:
```
notification {
   type = "Webhook"
   Url = "mywebsite.com"
}
```

etc.

Reply:
Yeah, I didn't want to get into a "variable number of commas" situation. Perhaps I'm just being conservative, but I'd rather invest in an improved `notification` resource than such a change. (But, to be fair, I started the change by going that direction.)